agent/php_user_instrument.c

```diff
@@ -138,17 +155,11 @@ static void nr_php_wrap_zend_function(zend_function* func,
                                       nruserfn_t* wraprec TSRMLS_DC) {
 #if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO \
     && !defined OVERWRITE_ZEND_EXECUTE_DATA /* PHP8+ */
-  if (ZEND_USER_FUNCTION != func->type) {
-    nrl_verbosedebug(NRL_INSTRUMENT, "%s%s%s is not a user function",
-                     wraprec->classname ? wraprec->classname : "",
-                     wraprec->classname ? "::" : "", wraprec->funcname);
-
-    /*
-     * Prevent future wrap attempts for performance and to prevent spamming the
-     * logs with this message.
-     */
-    wraprec->is_disabled = 1;
-    return;
+  if ((NULL != func) && (NULL != func->common.scope)) {
+    if ((NULL != func->common.scope->name)
+        && (NULL == wraprec->reportedclass)) {
+      wraprec->reportedclass = nr_strdup(ZSTR_VAL(func->common.scope->name));
+    }
```

Comment:
suggestion for abstraction to improve readability:
```
static bool chk_reported_class(zend_function* func, nruserfn_t* wraprec) {
  if ((NULL == func) || (NULL == func->common.scope)) {
    return false;
  }

  if ((NULL == func->common.scope->name) || (NULL != wraprec->reportedclass)) {
    return false;
  }
  return true;
}
```
```suggestion
  if (chk_reported_class(func, wraprec)) {
    wraprec->reportedclass = nr_strdup(ZSTR_VAL(func->common.scope->name));
  }
```

Reply:
Yes, more readable...but going back to your point about every additional function call adds to performance overhead since this gets called/executed very often.

Reply:
Sounds good, can you add this?